apps/examples/smartfs/fs_performance/fs_performance_main.c

```diff
@@ -0,0 +1,525 @@
+		printf("**********************************TEST GC EFFECT************************************");
+		printf("\n****************************************************************************************\n");
+		for (i = 1; i <= ITR_GC; i++) {
+			if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&before) < 0) {
+				fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+				ret = -1;
+				goto close_with_frt;
+			}
+
+			fd = open(g_file[0], O_WROK | O_CREAT);
+			if (fd < 0) {
+				printf("Unable to open file: %s, fd: %d\n", g_file[0], fd);
+				ret = ERROR;
+				goto close_with_frt;
+			}
+
+			ret = write(fd, g_rw_buf, g_fsz);
+			if (ret != (g_fsz)) {
+				 printf("Unable to write to file: %s, ret: %d\n", g_file[0], ret);
+				 ret = ERROR;
+				 goto close_with_fd;
+			}
+			close(fd);
+
+			ret = unlink(g_file[0]);
+			if (ret != OK) {
+				printf("Failed to unlink file: %s, errno: %d\n", g_file[0], errno);
+				ret = ERROR;
+				goto close_with_frt;
+			}
+
+			if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&after) < 0) {
+				fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+				ret = -1;
+				goto close_with_frt;
+			}
+
+			itr_time = after.timeleft - before.timeleft;
+			printf("iteration time: %lu us\n", i, itr_time);
+			tot_time += itr_time;
+		}
+		printf("\nFinal time: %lu us\n", tot_time);
+		break;
+
+	default:
+		break;
+	}
+
+	if (ioctl(frt_fd, TCIOC_STOP, 0) < 0) {
+		fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+		ret = -1;
+		close(frt_fd);
+	}
+
+	close(frt_fd);
+	return ret;
+
+close_with_fd:
+	close(fd);
+close_with_frt:
+	if (ioctl(frt_fd, TCIOC_STOP, 0) < 0) {
+		fprintf(stderr, "ERROR: Failed to stop the Free Run Timer: %d\n", errno);
+	}
+
+	close(frt_fd);
+
+	return ret;
+}
+
+static void show_usage() {
+	printf("\nusage: fs_performance [Timer device no.] [Test case] [File size(bytes)] [Buffer size(bytes)]\n");
+	printf("Timer Device no. :         FRT Timer device\n");
+	printf("Test Case        :         Test case to be executed\n");
+	printf("      1. File Creation Test\n");
+	printf("      2. Overwrite Test\n");
+	printf("      3. Append Test\n");
+	printf("      4. Read Test\n");
+	printf("      5. Garbage Collection Effect Test\n");
+	printf("File Size        :         File size(bytes) less than equal to 64kb\n");
+	printf("Buffer Size      :         Buffer size(bytes) less than equal to File size\n");
+}
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int fs_performance_main(int argc, char *argv[])
+#endif
+{
+
+	int option = -1;
+	int ret = 0;
+	int timer_devno = 0;
+	int selected_opt;
+	g_fsz = MAX_FILE_SIZE;
+	g_bsz = MAX_RD_BUF_SIZE;
+
+	if (argc == 2 && !strncmp(argv[1], "--help", strlen("--help") + 1)) {
+		show_usage();
+		return OK;
+	}
+
+	if (argc > 4) {
+		timer_devno = atoi(argv[1]);
+		option = atoi(argv[2]);
+		if (option < 1 && option > 5) {
```

Comment:
I think this check should be 
`if (option < 1 || option > 5) `

Reply:
Thankyou. I too realized it.